var/spack/repos/builtin/packages/gnutls/package.py

```diff
@@ -36,6 +39,9 @@ class Gnutls(AutotoolsPackage):
     # gnutls+guile is currently broken on MacOS.  See Issue #11668
     conflicts("+guile", when="platform=darwin")
 
+    # newer versions of clang are not compatible with gnutls <= 3.7
+    conflicts("%clang@15:", when="@:3.7")
```

Comment:
I can't reproduce. Only thing is that it sets a bunch of `-W` warning flags that are gcc only, but that's not an error in clang.